store/sql_webhook_store.go

```diff
@@ -524,3 +536,92 @@ func (s SqlWebhookStore) AnalyticsOutgoingCount(teamId string) StoreChannel {
 
 	return storeChannel
 }
+
+func (s SqlWebhookStore) SaveCommand(webhook *model.CommandWebhook) StoreChannel {
+	storeChannel := make(StoreChannel, 1)
+
+	go func() {
+		result := StoreResult{}
+
+		if len(webhook.Id) > 0 {
+			result.Err = model.NewLocAppError("SqlWebhookStore.SaveCommand",
+				"store.sql_webhooks.save_command.existing.app_error", nil, "id="+webhook.Id)
+			storeChannel <- result
+			close(storeChannel)
+			return
+		}
+
+		webhook.PreSave()
+		if result.Err = webhook.IsValid(); result.Err != nil {
+			storeChannel <- result
+			close(storeChannel)
+			return
+		}
+
+		if err := s.GetMaster().Insert(webhook); err != nil {
+			result.Err = model.NewLocAppError("SqlWebhookStore.SaveCommand", "store.sql_webhooks.save_command.app_error", nil, "id="+webhook.Id+", "+err.Error())
+		} else {
+			result.Data = webhook
+		}
+
+		storeChannel <- result
+		close(storeChannel)
+	}()
+
+	return storeChannel
+}
+
+func (s SqlWebhookStore) GetCommand(id string) StoreChannel {
+	storeChannel := make(StoreChannel, 1)
+
+	go func() {
+		result := StoreResult{}
+
+		var webhook model.CommandWebhook
+
+		exptime := model.GetMillis() - model.COMMAND_WEBHOOK_LIFETIME
+		if err := s.GetReplica().SelectOne(&webhook, "SELECT * FROM CommandWebhooks WHERE Id = :Id AND CreateAt > :ExpTime", map[string]interface{}{"Id": id, "ExpTime": exptime}); err != nil {
+			result.Err = model.NewLocAppError("SqlWebhookStore.GetCommand", "store.sql_webhooks.get_command.app_error", nil, "id="+id+", err="+err.Error())
+			if err == sql.ErrNoRows {
+				result.Err.StatusCode = http.StatusNotFound
+			}
+		}
+
+		result.Data = &webhook
+
+		storeChannel <- result
+		close(storeChannel)
+	}()
+
+	return storeChannel
+}
+
+func (s SqlWebhookStore) TryUseCommand(id string, limit int) StoreChannel {
+	storeChannel := make(StoreChannel, 1)
+
+	go func() {
+		result := StoreResult{}
+
+		if sqlResult, err := s.GetMaster().Exec("UPDATE CommandWebhooks SET UseCount = UseCount + 1 WHERE Id = :Id AND UseCount < :UseLimit", map[string]interface{}{"Id": id, "UseLimit": limit}); err != nil {
+			result.Err = model.NewLocAppError("SqlWebhookStore.TryUseCommand", "store.sql_webhooks.try_use_command.app_error", nil, "id="+id+", err="+err.Error())
+		} else if rows, _ := sqlResult.RowsAffected(); rows == 0 {
+			result.Err = model.NewLocAppError("SqlWebhookStore.TryUseCommand", "store.sql_webhooks.try_use_command.invalid.app_error", nil, "id="+id)
```

Comment:
You can use `NewAppError` here . It has the same parameters as `NewLocAppError`, but it also takes the status code as its last argument, so it saves you a line